test/e2e/common/util_cleanup.go

```diff
@@ -173,32 +249,40 @@ func DeleteAllPools() {
 	if numPools != 0 {
 		logf.Log.Info("DeleteAllPools: ", "Pools", pools.Items)
 	}
+	return numPools == 0
 }
 
+//  >=0 definitive number of mayastor pods
+// < 0 indeterminate
 func MayastorUndeletedPodCount() int {
-	pods, err := gTestEnv.KubeInt.CoreV1().Pods("mayastor").List(context.TODO(), metav1.ListOptions{})
+	ns, err := gTestEnv.KubeInt.CoreV1().Namespaces().Get(context.TODO(), NSMayastor, metav1.GetOptions{})
 	if err != nil {
-		logf.Log.Error(err, "MayastorUndeletedPodCount: list pods failed.")
+		logf.Log.Error(err, "MayastorUndeletedPodCount: get namespace")
+		return -1
+	}
+	if ns == nil {
+		// No namespace => no mayastor pods
 		return 0
 	}
-	if pods != nil {
-		return len(pods.Items)
+	pods, err := gTestEnv.KubeInt.CoreV1().Pods("mayastor").List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		logf.Log.Error(err, "MayastorUndeletedPodCount: list pods failed.")
+		return -1
 	}
-	logf.Log.Info("MayastorUndeletedPodCount: nil list returned.")
-	return 0
+	return len(pods.Items)
 }
 
 // Force deletion of all existing mayastor pods
-// Returns true if pods were deleted, false otherwise
-func ForceDeleteMayastorPods() bool {
+// Returns true if pods were deleted, false otherwise,
```

Comment:
Should this comment be "returns true if any pods were found" ?

Reply:
Depends, ForceDeleteMayastorPods deletes any pod it finds, that is it raison d'etre